src/main/java/io/cdap/plugin/salesforce/plugin/sink/batch/SalesforceRecordWriter.java

```diff
@@ -93,16 +92,15 @@ private void submitCurrentBatch(boolean startNewBatch) throws IOException {
       } catch (AsyncApiException e) {
         throw new RuntimeException("There was issue communicating with Salesforce", e);
       }
-    }
-
-    if (startNewBatch) {
-      csvBuffer = new CSVBuffer(true);
+      csvBuffer.reset();
     }
   }
 
   @Override
   public void close(TaskAttemptContext taskAttemptContext) throws IOException {
-    submitCurrentBatch(false);
+    submitCurrentBatch();
+    csvBufferSizeCheck.close();
```

Comment:
can these throw IOException?

If so, we need to catch exceptions and make sure we attempt to close both. Also need to make sure that even if these throw an exception we still wait for the job to complete. Probably simpler if these are closed after waiting for the job.

Reply:
done